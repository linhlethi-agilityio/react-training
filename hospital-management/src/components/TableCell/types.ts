import { IData } from '@type/data';
import { ITableColumn } from '../CustomTable/types';

export interface ITableCellProps<T extends IData> {
  item: T;
  columnConfig: ITableColumn<T>;
}
