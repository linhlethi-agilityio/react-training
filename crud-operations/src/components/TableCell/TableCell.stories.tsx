import type { ComponentStory, Meta } from '@storybook/react';
import { Table, Tbody, Tr } from '@chakra-ui/react';

// Components
import TableCell from '.';

const Template: ComponentStory<typeof TableCell> = (args) => (
  <Table>
    <Tbody>
      <Tr>
        <TableCell {...args} />
      </Tr>
    </Tbody>
  </Table>
);

const meta: Meta<typeof Table> = {
  component: Template as any,
};

export default meta;

export const Default = Template.bind({});
const props = {
  children: 'Default',
};
Default.args = props;
