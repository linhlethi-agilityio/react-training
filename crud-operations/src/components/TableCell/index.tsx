import { ReactNode } from 'react';
import { Td } from '@chakra-ui/react';

interface TableCellProps {
  children: ReactNode;
}

const TableCell = ({ children }: TableCellProps) => {
  return (
    <Td fontSize="text.sm" lineHeight="sm" color="default">
      {children}
    </Td>
  );
};

export default TableCell;
