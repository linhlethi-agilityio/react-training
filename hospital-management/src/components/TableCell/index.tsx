import { ReactNode } from 'react';

// Types
import { ITableCellProps } from './types';
import { IData } from '@type/data';
import { ITableColumn } from '../CustomTable/types';

// Hocs
import { typedMemo } from '@hocs';

const TableCell = <T extends IData>(props: ITableCellProps<T>) => {
  const { item, columnConfig } = props;

  const renderCell = (item: T, columnConfig: ITableColumn<T>): ReactNode => {
    const { accessor } = columnConfig;

    if (typeof accessor === 'string') {
      return item[accessor]?.toString();
    }

    if (typeof accessor === 'function') {
      return accessor(item);
    }
  };

  return <td>{renderCell(item, columnConfig)}</td>;
};

export const MemoizedTableCell = typedMemo(TableCell);
