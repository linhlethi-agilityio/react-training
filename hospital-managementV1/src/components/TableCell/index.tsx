import styled from 'styled-components';
import { ITableCellProps } from './types';

const StyledTableCell = styled.td<ITableCellProps>`
  width: ${(props) => props.cellWidth};
  text-align: ${(props) => props.cellTextAlign};
  font-weight: var(--font-weight-normal);
  font-size: var(--text-xx-small);
  line-height: 20px;
  padding: 18px 0;
  vertical-align: middle;
`;

export { StyledTableCell };
