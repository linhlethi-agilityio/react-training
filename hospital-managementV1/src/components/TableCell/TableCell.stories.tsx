import { ComponentMeta, ComponentStory } from '@storybook/react';
import { StyledTableCell } from '@components/TableCell';

export default {
  title: 'TableCell',
  component: StyledTableCell,
} as ComponentMeta<typeof StyledTableCell>;

const Template: ComponentStory<typeof StyledTableCell> = (args) => <StyledTableCell {...args} />;

export const Default = Template.bind({});
const props = {
  children: 'Default',
};
Default.args = props;
