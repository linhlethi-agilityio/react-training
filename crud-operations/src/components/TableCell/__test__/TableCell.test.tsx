import { render } from '@testing-library/react';
import { Table, Tbody, Tr } from '@chakra-ui/react';

// Components
import TableCell from '..';

describe('TableCell', () => {
  test('should match snapshot for TableCell', () => {
    const { container } = render(
      <Table>
        <Tbody>
          <Tr>
            <TableCell>inches</TableCell>
          </Tr>
        </Tbody>
      </Table>,
    );

    expect(container).toMatchSnapshot();
  });
});
