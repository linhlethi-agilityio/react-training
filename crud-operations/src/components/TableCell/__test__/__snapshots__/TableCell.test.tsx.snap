// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell should match snapshot for TableCell 1`] = `
<div>
  <table
    class="chakra-table css-0"
  >
    <tbody
      class="css-0"
    >
      <tr
        class="css-0"
      >
        <td
          class="css-8l52he"
        >
          inches
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
